src/main/scala/com/evolutiongaming/conhub/ConHub.scala

```diff
@@ -26,7 +26,8 @@ object ConHub {
 
     def conStates: ConStates[Id, T, M]
 
-    def cons(l: L, localCall: Boolean = true): Iterable[C] = searchEngine(l, conStates.values, localCall)
+    def cons(l: L, localCall: Boolean = true): Iterable[C] =
+      Option(searchEngine).toIterable.flatMap(_.apply(l, conStates.values, localCall))
```

Comment:
Personally I don’t like the fix, why would you pass null here? Maybe we should re-write initialisation phase?

Reply:
Returning empty list does not violate, however passing `searchEngine` as `null` - does

Reply:
So, as I understand, you're proposing to change design of `ConHubImpl` to avoid `val (searchEngine, conStates, sendMsgs) = connect(onMsgs)` (`onMsgs` calls `searchEngine` indirectly)

Reply:
yes, for instance.